ocaml/typing/ctype.ml

```diff
@@ -3828,7 +3828,8 @@ let filter_arrow env t l ~force_tpoly =
               (* CR layouts v5: Change the Layout.value when option can
                  hold non-values. *)
               (Tconstr(Predef.path_option,
-                       [newvar2 level (Layout.value ~why:Type_argument)],
+                       [newvar2 level (Layout.value ~why:(Imported_type_argument
```

Comment:
This layout has to be used a bunch. Probably best to export it from predef, maybe as `option_argument_layout`. That will also make it easy to revise when we allow `type ('a : any) option`, which won't be all that far in the future.